File: android/src/main/kotlin/com/example/nearby_cross/callbacks/ConnectionCallbacks.kt
package com.example.nearby_cross.callbacks

abstract class ConnectionCallbacks {
    abstract fun onPayloadReceived(bytesReceived: ByteArray, endpointId: String)
    abstract fun onDisconnected(endpointId: String)
    abstract fun onConnectionInitiated(
        endpointId: String,
        endpointName: ByteArray,
        alreadyAcceptedConnection: Boolean
    )

    abstract fun onSuccessfulConnection(endpointId: String)

    abstract fun onRejectedConnection(endpointId: String)
}

File: ios/Classes/callbacks/ConnectionCallbacks.swift
//
//  ConnectionCallbacks.swift
//  nearby_cross
//
//  Created by Ignacio Velasco on 29/02/2024.
//

import Foundation

protocol ConnectionCallbacks {
    func onPayloadReceived(bytesReceived: Data, endpointId: String)
    func onDisconnected(endpointId: String)
    func onConnectionInitiated(endpointId: String, endpointName: Data, alreadyAcceptedConnection: Bool)
    func onSuccessfulConnection(endpointId: String)
    func onRejectedConnection(endpointId: String)
}
